libafl/src/observers/cmp.rs

```diff
@@ -348,6 +349,263 @@ struct cmp_map {
 };
 */
 
+/// A [`CmpObserver`] observer for AFL++ redqueen
+#[derive(Serialize, Deserialize, Debug)]
+pub struct AFLStdCmpObserver<'a, S>
```

Comment:
Why a new observer?

Reply:
It is because it appends different types of metadata

I can put all this in StdCmpObserver, only append_metadata() is different after all

Reply:
I think I need this new observer after all.
otherwise specialization is needed (but then afl-rq will need nightly)

so, this observer is special for AFLCmpMap https://github.com/AFLplusplus/LibAFL/blob/rq/libafl/src/observers/cmp.rs#L366.
the reason is here
https://github.com/AFLplusplus/LibAFL/blob/rq/libafl/src/observers/cmp.rs#L418
I need to get header from the cmpmap here.

but I can't implement header(&self, idx: usize) for trait `CmpMap` because the type is different depending on the map.
Current libafl cmplog header is this guy, 32bit
```
pub struct CmpLogHeader {
    hits: u16,
    shape: u8,
    kind: u8,
}
```
but AFL++ cmplog header is more complex, and it's [u8, 8]